Gas Mask Variety Take Two (Not Summoning Every Maintainer Edition)

Paxilmaniac opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Adds three variants of the gas mask, replaces fire locker gas masks with a random pick between two of the new variants.

![image](https://user-images.githubusercontent.com/57122098/72036756-b4c80000-3250-11ea-93ca-e46ad4e19fab.png)

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
Gas masks are a go to mask, because they both hide your identity AND can be used as internals. This doesn't remove the old gas masks, but makes them much harder to find, and replaces the most common sources with one of three new variants that makes trade offs between each other, so there are less of the all powerful greytide masks everywhere. First this and then i'll make gas masks actually work like gas masks ok??

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
add: Added three new gas mask variants; one which doesn't hide identities or block gases, one which doesn't hide eyes or block gases, and a final one which is purely aesthetic & can't be used as internals, nor blocks gases.
balance: fire locker gas masks are now chosen randomly between the first two new ones, as well as a normal gas mask.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

carda-sometimesknownas-paxilmaniac committed:
what is similar about the last pr and sybil, both are dead and bad

take two ok here we go and yes i probably should have asked orange man for help instead of cavemanning it

carda-sometimesknownas-paxilmaniac committed:
dont know why the swat mask was in there but ok

very cool github

wesoda25 commented:
just gonna take this from the old one

That comment doesn't even make sense though, how does it make sense to make gas masks, the thing you are supposed to have when fighting fire, not be able to be connected to internals? I don't even get whats "good" about the mask, its a liable reason to be arrested by shitsec so any real advantage it provides is ???? This PR seems like pointless bloat tbh

ShizCalev commented on code/modules/clothing/masks/gasmask.dm:
```diff
@@ -11,6 +11,39 @@
+	desc = "A face-covering mask that cannot be connected to an air supply. Turns out to not be that great at blocking gasses."
+	icon_state = "gas_alt4"
+	armor = list("melee" = 0, "bullet" = 0, "laser" = 0,"energy" = 0, "bomb" = 0, "bio" = 10, "rad" = 10, "fire" = 5, "acid" = 5)
+	clothing_flags = BLOCK_GAS_SMOKE_EFFECT
```
i concur with the sentiments in the other thread honestly, having a gas mask that appears in an emergency fire closet that can't actually be used as internals is pretty shitty. If you want to conceal your identity, there's plenty of bandanas and crap laying around the station already that will do that for you.

carda-sometimesknownas-paxilmaniac committed:
moves the non internals mask to maint loot

i mean yeah i guess you have a point, probably going to add it back to lockers if i add gas masks actually filtering gasses though

Shadowflame909 commented:
It being a bunch of new sprite changes instead of a rebalance would be pretty great

Ohagi-Chan commented:
"replaces the most common sources with one of three new variants that makes trade offs between each other"

Can we get a description of their differences in the pr description?
Blink and you might accidentally assume this is purely a sprite change.

ShizCalev commented:
Edited your changelog to be a little more descriptive.

MrPerson commented:
I'm inclined to agree with everyone else that just the sprite changes would have been fine and very welcome.

Qustinnus commented on code/modules/clothing/masks/gasmask.dm:
```diff
@@ -11,6 +11,39 @@
+
+/obj/item/clothing/mask/gas/alt3
+	name = "gas mask"
+	desc = "A face-covering mask that cannot be connected to an air supply. Turns out to not be that great at blocking gasses."
```
description does not match flags

Michael-Ashfield commented:
Do the non-internals ones at least filter toxins from the air so you're not huffing hot plasma through a filtered gasmask while fighting a plasma fire?

Dorsisdwarf commented:
I dont like the idea of putting a completely worthless fake gas mask on the station to trick people into dying.

TheVekter commented:
Yeah, no, this reeks of ided and seems very disingenuous. I see no problem making these randomly spawn around the station, as they're good sprites (and the clear mask looks _much_ closer to a normal fireman's mask) but having them functionally different is just unnecessary.

Tlaltecuhtli commented:
is it really that hard to just add different gas mask sprites without having to add """balance changes""""

AWalton commented:
I don't see the point in having a gas mask which can't be used as internals, even if it isn't found in fire lockers and only in maint. 

There are a ton of different masks that people can wear to hide their identity instead. Also some helmets I believe don't hide identity like cardborg and laser tag helmet which should be rebalanced to do so as well. 

Many masks which shouldn't intuitively work as internals like the clown and mime mask do in fact, because why would anyone ever opt for a non-internals mask when they're needed so frequently and internals masks are so commonplace. 

And having a non-internals gas mask is just going to get people killed when they throw away their breath mask unknowingly and get caught in a gas flood or depressurised area.

Iatots commented:
I WISH people cared about random unknowns wearing gasmasks and their work attire - ID.

TheVekter commented:
> 
> 
> I WISH people cared about random unknowns wearing gasmasks and their work attire - ID.

Off-topic but I always ask if I'm working sec.

Paxilmaniac commented:
fuck it i might as well make gas masks filter gasses like actual gas masks while im at it
